### DISCUSSION TITLE
posts/why-i-chose-the-thinkpad-z13-as-my-linux-laptop/

### DISCUSSION THREAD

**Discussion by giscus** [flexiondotorg/flexiondotorg.github.io#195] [Blog Comments]
# posts/why-i-chose-the-thinkpad-z13-as-my-linux-laptop/

Blog, Podcasts, Live Streams and Videos about Linux and Open Source with a focus on Ubuntu, Debian and NixOS.

https://wimpysworld.com/posts/why-i-chose-the-thinkpad-z13-as-my-linux-laptop/

**Comment by adingbatponder** [comment]
A big thank you for this. I listened to the podcast also. I found that your criteria were exactly right and really helped me to be more confident about which parameters to consider. It lead me to think about one aspect that is missing when people review laptops. What does a desktop with essentially the same specs cost? I guess it would come out a bit odd sometimes, but it strikes me that the "same spec desktop device" would be an interesting way of showing the reader what is so good about laptops and what is best to do if working on the go is not required. All the best.

**Reply by flexiondotorg** [reply]
Glad you found the write up useful and thanks for listening to [Linux Matters](https://linuxmatters.sh) 😀

**Comment by julian-alarcon** [comment]
A lot of GPU reported bugs for AMD Ryzen. Did you found any of those?
https://wiki.archlinux.org/title/Lenovo_ThinkPad_T14s_(AMD)_Gen_3

**Reply by flexiondotorg** [reply]
I've not experienced anything like what is described in that page.

**Comment by jhit** [comment]
Hello, 
I own a Lenovo P1Gen4 with Nvidia T1200. I am so frustrated by the fan noise. Even on light usage the two fans are always active and spinning at around 3000rpm.

How is the Z13 doing in this regard? Does the fan go off most of the time?

**Reply by OrlandoRodriguez-KSO** [reply]
Hello, I do not feel that the fans are noisy in this laptop (unfortunately I do not have measurements). What I could say is it could be pretty hot in some cases like playing games, however, the performance keeps pretty well even on that situation. About performance, battery and fans I have no regrets with this laptop.

**Reply by jhit** [reply]
Hey @orrot do the fans run all the time or do they completely stop on light tasks like browsing web and writing email?

**Reply by OrlandoRodriguez-KSO** [reply]
Hi @jhit, I have been checking the fans using lm-sensors, it looks like they stop and do not need to work if we are doing light task and the laptop is in a good place (in a table or in the support when I usually work). However, If I am on the bed, with the laptop on the mattress, the fans start to work at high RPMs according to lm-sensors. At this moment, writing this post, with 10 tabs opened and a youtube video in the background, lm-sensors is printing 0 RPM.

**Reply by flexiondotorg** [reply]
I don't think I've ever heard the fans spin up on the Z13. I know the fans are active sometimes, but I've not heard them and not been distracted by them.

**Comment by OrlandoRodriguez-KSO** [comment]
Very nice text! I have pretty similar requirements (except that I was not able to import a 32 Ram model to my country). I have been using for several months the laptop, with KDE Neon and now with Ubuntu, and both experiences were great!. Fingerprint worked for me in all the cases, I just had issues to disable it for login in Ubuntu (I enjoy fingerprint and IR for facial recognition for all the other things that require password, but not for login because of the keyring). Sound is great, battery is great, performance is great and even, some light gaming is great for a laptop without eGPU. I had to use newer Kernels to get the laptop working well, and to be honest I had several issues, until I get Kernel 6.2 where all finally was solved (I am using amd pstate drivers to keep a good battery life with amd). The only thing I am not happy about this laptop is that sometimes a pop sound is happening. The issue is reported by in the lenovo page, but no solution has been provided. There is also another reported issue for power profiles, but it looks it will be released a solution for that in the next months.

**Comment by sergiusens** [comment]
Hey Martin, I am really interested in this laptop. I gave my current one away and now typing from that Dell XPS13 you once gave to me 🙏, and it is sloooow, which I was personally surprised about. I am urgently looking for a new laptop, I have fallen to think I want this z13 or one of those newer Dell XPS 13 Pluses (the Dell looks are just too nice and I am a sucker for it).

I do have a a question about this z13, I have not seen many people with success stories when using opencl with AMD graphics, have you had a play with that and does it work ok? (I am thinking Darktable here)

**Reply by flexiondotorg** [reply]
Hey Sergio 👋 I haven't don't anything with OpenCL in the Z13 so sadly have helpful to add on that topic. I would say the current generations of AMD CPUs (6000 and 7000 series) are performance competitive with Intel's offerings, but the AMD CPUs are way more energy efficient, and therefore have better battery endurance and output less heat. The iGPs in the current AMD CPUs are also significantly better than anything Intel offer.

I expect the the Z13 will perform very well with Darktable.

**Comment by poser** [comment]
Thank you for the thorough and helpful review! Your criteria are my criteria. One thing, though... I'm a TrackPoint user, and the last time I tried to work without a row of physical buttons below the pointing stick, I ended up sad. (1st gen Thinkpad Helix maybe?) I typically disable the touchpad entirely because palm rejection was less than perfect last time I tested it under Linux. And/or I'm a sloppy typist. Even with the touchpad enabled, though, I found the drivers insufficient to meet the following requirements when using a device that simulates TrackPoint buttons using the upper portion of the touchpad:

- I need middle-click paste to work, 
- I need middle-click scrolling to work with the TrackPoint, 
- I need it not to do the former when I'm trying to do the latter, and
- I need an accidental brush of the trackpad to do nothing at all.

Any chance you're a TrackPoint user? Or tested these things just because you were super bored? :)

**Reply by flexiondotorg** [reply]
Sadly not. I've never used the Trackpoint on the Z13. But @popey is a Z13 owner and traditionally a trackpoint advocate 🙂 Perhaps he has some insight?

**Reply by popey** [reply]
I am indeed a TrackPoint™ user. However, my Z13 is mostly a desktop-replacement, connected to a USB C hub with an external keyboard and mouse. It does indeed have a functional set of 3 'buttons' to use with the TrackPoint™ but because (unlike the classic X220) the buttons aren't separate, but one bar with three regions, it is easy to hit the wrong button. HTH YMMV

**Reply by poser** [reply]
Thank you!

**Comment by tstechy** [comment]
Have you already posted about your after-market solution to upgrade to 2TB?

**Reply by flexiondotorg** [reply]
Not yet. Coming soon ™️

**Comment by stevemoon** [comment]
After some digging I finally discovered why I can't buy the same laptop you describe: Lenovo in USA only offers a Linux option on one laptop. The Z13 available for the USA market has a different CPU, different screen, etc. <head exploding> 

Thanks for the podcast and writeup though!

**Comment by popey** [comment]
I did a similar SSD upgrade that Martin did, but hasn't got around to writing up yet :)

https://popey.com/blog/2023/09/thinkpad-z13-ssd-replacement/

**Reply by flexiondotorg** [reply]
@tstechy That blog post 👆️ from Alan is all the info you need on how to upgrade the Z13 SSD.

The only additional information for me to add is the 2TB SSD I used is a [Western Digital PC SN740 NVMe™ 2TB M.2 2230 SSD](https://www.westerndigital.com/en-gb/products/internal-drives/pc-sn740-ssd#SDDPTQE-2T00). I got mine on eBay.

**Comment by KevTett** [comment]
Thanks for this post. I'm creating a similar spreadsheet with my requirements for a Linux laptop. :-)
Do you know of any resources that supply lists of current laptops and allow filtering by features - a bit like the Kimovil site does for mobiles?

**Reply by flexiondotorg** [reply]
Sorry, I don't :disappointed:

**Comment by fraricci** [comment]
Hi! Thanks a lot for sharing this precious data on different laptops. I wanted to ask if after more than a year you had any issue with the Z13. In particular, looking at this wiki https://wiki.archlinux.org/title/Lenovo_ThinkPad_Z13, it looks like some of the hardware is partially working (audio and usb4 in particular). What's your experience? Thanks!

**Reply by OrlandoRodriguez-KSO** [reply]
Hi @fraricci, I just make a basic usage of the USB4 ports. I have been using several adapters and ports without any issue. One of them is a dock with a couple of DisplayPort and HMI plus several other ports like USBC3.1, USBA 3.0, Ethernet and 3.5mm audio. The dock has worked very well for me. I have not tried eGPU or things like that, but at least related to docks, hubs and adapters, my experience have been pretty good, specially in Kernels 6.X.

**Comment by darkhatplatypus** [comment]
Hi! Love your blog post :) I have the same laptop with 16GB/512GB and touch screen option and been loving it so far. I want to ask if you face any issues with Linux compatibility as well? I'm running Fedora 40, and sometimes the fingerprint just stops working after a while (more info here https://gitlab.freedesktop.org/libfprint/libfprint/-/issues/612) and the laptop would sometimes wake up during sleep and is stuck on a black screen requiring a hard reset. 
I also got this laptop due to the insane battery claims I saw around, but only got 7-8 hours of real work usage (a couple of Firefox windows, VSCode, brightness on ~30%) even on light work (just typing in google docs). Is there something else I need to do? I'm not familiar with optimizing power management on linux systems and just using the default options for now.
Thank you very much!

**Reply by flexiondotorg** [reply]
I've not experienced any issues with the finger print reader, in fact I've been rather impressed with it's reliability and speed. I have experience some issues with the amdgpu driver, specifically the laptop not being able to resume from suspend with Linux 6.7 and 6.8. I believe this issue is resolved in 6.9 but I have found Linux 6.6 to be reliable and therefore that is the kernel version I have elected in my NixOS configuration as it is an LTS kernel.

As for battery endurance, I'd say 7 - 8 hours is typical for me these days. I haven't invested any effort is tuning power profiles in NixOS. The longer 11 hour battery endurance is as a result of the pre-configured power tuning that was present in the Ubuntu factory factory image.

**Reply by OrlandoRodriguez-KSO** [reply]
Hi guys, In my case I have had the mentioned issues with my model. The one with Black screen after wake up was specially anoying some time ago. I had to put a script during suspend/wakeup, to turn the  WIFI off with "rfkill" (I am using Fedora) to avoid the suspend issue and I also had a problem with freezing screen with iGPU drivers and I had to use a kernel parameter (amdgpu.dcdebugmask=0x4) to avoid the "freezed display" problem. That was some time ago, currently I don't know if those problems are still there because I have upgraded a couple of times my system, but I haven't tested again without the fixes I applied (My laptop is working well so I keep the configuration, if you need details about what I did, just let me know).

About the battery, I was having pretty similar battery than yours (6-7 hours), however, watching youtube videos it was wasting a lot power (12w approx). I was testing several things to improve this (checking other browsers for HW acceleration, different power profiles, different scaling drivers for the CPU like pstate active vs passive, etc). After several tests, for some reasons I don't know, in my laptop the best configuration I found in Fedora is to use any of the scaling drivers: amdpstate active or passive (with some differences in configuration, however, powerprofiles daemon in Fedora controls this very well by itself) and I changed my Kernel to an Arch distro Kernel with name "cachyOS". That Kernel with UKSMD improved the battery to 8-9w for youtube videos, and of course, web browsing or other activies can spend less power. In my regular usage, the battery increased to 8h-10h. That is the configuration I have now, not sure if it could work for you guys, but well, that worked for me. I am using Fedora 40 with Gnome.

**Reply by darkhatplatypus** [reply]
@flexiondotorg thanks, that might be an issue with Fedora alone, unfortunately Lenovo doesn't provide official Ubuntu restore images so those numbers are sort of lost to time :( I don't know if there's a problem with my unit but my experience with the fingerprint reader seems quite unreliable, as it requires multiple scans to get right and in many cases falls back to passwords.

@orrot I just got the aforementioned black screen after wakeup with kernel 6.9.? and BIOS 1.66 as of last week. I upgraded to BIOS 1.67 and will report back if the issue persists. It's very annoying because it happens so rarely. Can you share your rfkill script? That would be very helpful.

Relating to HW acceleration, I had to use mesa-va-drivers-freeworld as the default drivers lack H264 hardware decoding support and watching a Twitch stream on battery would eat up 17W(!). I don't know how the Windows drivers got it down to 5W when playing in a Chromium browser (Edge on Windows). That really tempted me to come back to Windows :( My laptop hovers around 6-7W playing YouTube (both in VP9 and AV1 w/ hardware acceleration), which is pretty decent, but anything like scrolling to the comments jumps up to 10W. I feel like the power management isn't aggressive enough without going to Power Saver mode. I'll look through your configuration to see if there's anything I can try. Thank you very much.

**Reply by flexiondotorg** [reply]
> @flexiondotorg thanks, that might be an issue with Fedora alone, unfortunately Lenovo doesn't provide official Ubuntu restore images so those numbers are sort of lost to time :( I don't know if there's a problem with my unit but my experience with the fingerprint reader seems quite unreliable, as it requires multiple scans to get right and in many cases falls back to passwords.

Have you updated the firmware on your Z13? I think I recall at least one update being biometric related. Might be worth looking into.

**Reply by OrlandoRodriguez-KSO** [reply]
@droussielle With my current configuration, I also can watch several youtube videos between 6-7w, I woul say that is the average for me, but the channels I usually follow have 60fps and that increase a bit the consumption, to be honest I don't know why some videos seems to be lighter than others even when they look even better (I am using VP9 almost all the time). I have been using pstate-passive drivers for long time (they are not the default in the Linux kernel) and the way as they work, the power consumption of average tasks does not change too much when using powersave profile or performance profile, because the profiles only limit the maximum amount of power that the laptop can consume (this is the reason why with passive drivers I keep "performance" governor all the time and I keep the laptop platform profile (power daemon) in "balanced" and not "powersave"). Again, I am not using the regular Fedora kernel, but cachyOS with uksmd.

About the rfill, my problem with black screen was caused by the Qualcomm Wifi card included in this laptop. I discovered the issue because if the Wifi is turned off, the black screen issue does not happen. So what I did to automate this process is the following process:

I created a systemd config by creating a file in: 

/etc/systemd/system/root-suspend-fix.service

[Unit]
Description=Suspend fix for ath11k_pci
Before=sleep.target
StopWhenUnneeded=yes

[Service]
Type=oneshot
RemainAfterExit=yes
ExecStart=bash -c "rfkill block wifi"
ExecStop=bash -c "sleep 3 && rfkill unblock wifi"

[Install]
WantedBy=sleep.target

rfkill is the command to turn the wifi off and on in Fedora. I have been using this config for several months (Probably more than 1 year), not sure if it is still required, but it has been working welll so far. I knew about this problem by some forums in Lenovo.

I also had another issue with Gnome: the Gnome session was crashing randomly, but that also was solved by the cachyOS kernel. That kernel in my particular case has been very helpful, not only because of the improvements in battery but also making more stable my laptop with Fedora. Currently I love this laptop a lot, but aquiring this laptop when it was pretty new model was a headache for Linux.

**Reply by darkhatplatypus** [reply]
@flexiondotorg I went back to Windows and updated everything, and it just happened again as I'm typing this. fprintd just died and any attempt at restarting it results in `fprintd[167213]: Ignoring device due to initialization error: endpoint stalled or request not supported` until full system restart :( I'll ask around on the Lenovo forums to see if it's any help. BTW can you update the original post since Lenovo released the haptic touchpad configuration tool? That would be very helpful :)

@orrot thank you very much :D , I'll poke around with performance settings and see if my laptop still crashes with the script :)

**Reply by darkhatplatypus** [reply]
@orrot can you test battery usage on Twitter using Firefox? I just noticed that my laptop would jump to 15W+ when loading Twitter (!) while everything else stays at ~5W. I am already using mesa-va-drivers-freeworld so Firefox would have H264 Hardware acceleration (you can check in about:support). I know the site is media heavy but why would it be so high?? Thanks in advance!

**Reply by OrlandoRodriguez-KSO** [reply]
> @orrot can you test battery usage on Twitter using Firefox? I just noticed that my laptop would jump to 15W+ when loading Twitter (!) while everything else stays at ~5W. I am already using mesa-va-drivers-freeworld so Firefox would have H264 Hardware acceleration (you can check in about:support). I know the site is media heavy but why would it be so high?? Thanks in advance!

I was checking Twitter and I got a similar behaviour (higher power consumption than other pages), however, I noticed that in Twitter all the videos were launched and autoplayed. I changed that configuration and power consumption improved a lot (but now I have to click to reproduce the videos...not that nice). 

I also was checking these days the "active" drivers (the default ones indeed in newer kernelr versions) because I was long time using the passive and they are now supported by the power daemon profile used in Fedora (and other distros too). Power Daemon changes all what is required automaticall and in fact, power consumption I think is better now on them than in passive.

**Reply by darkhatplatypus** [reply]
> > @orrot can you test battery usage on Twitter using Firefox? I just noticed that my laptop would jump to 15W+ when loading Twitter (!) while everything else stays at ~5W. I am already using mesa-va-drivers-freeworld so Firefox would have H264 Hardware acceleration (you can check in about:support). I know the site is media heavy but why would it be so high?? Thanks in advance!
> 
> I was checking Twitter and I got a similar behaviour (higher power consumption than other pages), however, I noticed that in Twitter all the videos were launched and autoplayed. I changed that configuration and power consumption improved a lot (but now I have to click to reproduce the videos...not that nice).
> 
> I also was checking these days the "active" drivers (the default ones indeed in newer kernelr versions) because I was long time using the passive and they are now supported by the power daemon profile used in Fedora (and other distros too). Power Daemon changes all what is required automaticall and in fact, power consumption I think is better now on them than in passive.

Thanks, I think it's something to do with the media on the website that makes it consumes more battery. I also found out about  how hardware decoding on AMD on Linux can actually consume more energy than software decoding (https://gitlab.freedesktop.org/drm/amd/-/issues/3195#note_2295146) and was able to replicate it with Chrome on my system. This is quite a bummer and part of a reason why media consumption on this platform is noticeably worse than on Windows :(

Media decoding aside, I do think that the amd_pstate driver is pretty good, I'm hovering 4W writing this on Firefox and averages 5W doing normal tasks (surfing the web, reading articles, etc.) and <3W on idle. I don't think it can go lower than that so for now I am satisfied with what this system offers :) Thank you everyone

**Reply by OrlandoRodriguez-KSO** [reply]
> Thanks, I think it's something to do with the media on the website that makes it consumes more battery. I also found out about how hardware decoding on AMD on Linux can actually consume more energy than software decoding (https://gitlab.freedesktop.org/drm/amd/-/issues/3195#note_2295146) and was able to replicate it with Chrome on my system. This is quite a bummer and part of a reason why media consumption on this platform is noticeably worse than on Windows :(
> 
I had noticed that about the hardware acceleration, and in fact, Firefox has not been always my main browser. However, I have to say that the real improvement of that AMD hardware acceleration comes with 4K or bigger videos. In those cases is when I've found that the consumption is a lot better with hardware acceleration enabled in Firefox. I think the same about the other tasks... the power consumption is really nice on this laptop and without having a big battery like other laptops. Maybe in the feature or in newer versions AMD or Linux itslelf improves the video streaming and that would be great for us. Luckily for me, not a big problem while on the other tasks are still working great (I am developer, so for my job, battery just works great). Thanks for the feedback!

**Reply by OrlandoRodriguez-KSO** [reply]
Hey Daniel, me again, I have been curious since we talked about battery consumption, so I have been reading some stuff. I just checked what power-daemon does and the only thing is not clear for me is why the  GPU performance level is still as 'auto' even in power mode. I have been doing some quick tests (no for long time) and it looks like setting the **power_dpm_force_performance_level** improves the battery consumption during video playback. I found this by reading but also in someone commented it in a forum. The person that commented said that it could makes sluttering in 4K60fps videos (I am testing it to see if that is right), however, for me looks like it works pretty well and in my case I see an noticeable improvement in videoplayback. Not sure if probably that's something you also would like to try. I will continue on this for the next days to see what problems could appear with that configuration.

The change is something like this:

echo 'low' | sudo tee -i /sys/class/drm/card0/device/power_dpm_force_performance_level > /dev/null

("auto" is the default value, so same command could be used to get the default config back )

**Reply by orrot** [reply]
@droussielle  After some days using the configuration I put in the comment above, definitely the battery is even better, not only for youtube videos, but for applications like spotify. I placed the configuration of "low" graphics when laptop is not plugged. (I am the same person of the comment above, but I have to apply some changes in my account)

**Reply by darkhatplatypus** [reply]
> @droussielle After some days using the configuration I put in the comment above, definitely the battery is even better, not only for youtube videos, but for applications like spotify. I placed the configuration of "low" graphics when laptop is not plugged. (I am the same person of the comment above, but I have to apply some changes in my account)

Hi Orlando, I tried your solution and indeed saw a significant drop in power use when setting GPU performance level to `low` ! It was hovering ~12W and dropped to ~10W just with that tweak (USB-C dock connected with an external monitor, 1080p YouTube video, Power Saver mode). I will try to see if gaming performance is affected, but so far 4K YouTube playback doesn't seem to be affected , so I'm happy with this tweak :)

Edit: After switching back to Balanced it jumps up to ~14W, so maybe there's still something that needs optimizing?

**Comment by darkhatplatypus** [comment]
Since this article was released Lenovo has provided a touchpad configuration tool for Linux users! You can get it here: [https://pcsupport.lenovo.com/us/en/products/laptops-and-netbooks/thinkpad-z-series-laptops/thinkpad-z13-type-21d2-21d3/downloads/ds561548-elan-haptic-pad-settings-tool-for-linux-thinkpad-z13-gen-1-z16-gen-1?category=Mouse,%20Touchpad,%20Keyboard%20and%20Pen](url)

![image](https://github.com/user-attachments/assets/8c0ae94d-2613-4448-8b74-dd640e9cbf52)

From my time using the laptop, I can say the haptic touchpad has worked really well and is the main reason I got this over a traditional X or T series Thinkpad. The clicking is quite different from that of a Macbook though, I would say the click on this one feels more like a sharp tactile click, than that of a Macbook, which has a mushy thump to it. With proper speed/acceleration profile this can feel right as home as a touchpad on a Macbook :) The Z13 Gen 2 may have a different click profile as it uses a Sensel touchpad instead of an Elan one, so I can't comment on that. I would really appreciate if anyone wants to share their experience!

**Comment by maxamillion** [comment]
I know this blog post is old at this point but I wanted to drop by and thank you for writing it anyways. I was considering picking up on of these second hand and I had some reservations about it that you covered. I have almost the exact same set of requirements you do so it was almost comedically perfect that I found your blog. I went ahead and ordered my new-to-me Z13 and can't wait for it to get here. Thanks again!

**Reply by flexiondotorg** [reply]
I'm glad this blog post was helpful :slightly_smiling_face:
